frontend/public/components/edit-yaml.jsx

```diff
@@ -766,14 +859,23 @@ export const EditYAML = connect(stateToProps)(
                     </div>
                   </div>
                 </div>
-                <ResourceSidebar
-                  isCreateMode={create}
-                  kindObj={model}
-                  height={height}
-                  loadSampleYaml={this.replaceYamlContent_}
-                  insertSnippetYaml={this.insertYamlContent_}
-                  downloadSampleYaml={this.downloadSampleYaml_}
-                />
+                {this.showSidebar(showSchema, showSamples, showSnippets) && (
+                  <ResourceSidebar
+                    isCreateMode={create}
+                    kindObj={model}
+                    height={height}
+                    loadSampleYaml={this.replaceYamlContent_}
+                    insertSnippetYaml={this.insertYamlContent_}
+                    downloadSampleYaml={this.downloadSampleYaml_}
+                    showSidebar={showSidebar}
+                    toggleSidebar={this.toggleSidebar}
+                    samples={samples}
+                    showSamples={showSamples}
```

Comment:
It's trivial enough for the component to check if these are empty. I wouldn't pass these through as flags, just check for empty arrays in the component.